Neos.Neos/Classes/Domain/Service/WorkspaceService.php

```diff
@@ -0,0 +1,349 @@
+use Neos\Neos\Domain\Model\UserId;
+use Neos\Neos\Domain\Model\UserInterface;
+use Neos\Neos\Domain\Model\WorkspaceClassification;
+use Neos\Neos\Domain\Model\WorkspaceDescription;
+use Neos\Neos\Domain\Model\WorkspaceMetadata;
+use Neos\Neos\Domain\Model\WorkspacePermissions;
+use Neos\Neos\Domain\Model\WorkspaceTitle;
+
+/**
+ * @api
+ */
+#[Flow\Scope('singleton')]
+final class WorkspaceService
+{
+    private const TABLE_NAME_WORKSPACE_METADATA = 'neos_neos_workspace_metadata';
+    private const TABLE_NAME_WORKSPACE_ROLE = 'neos_neos_workspace_role';
+
+    public function __construct(
+        private readonly ContentRepositoryRegistry $contentRepositoryRegistry,
+        private readonly PrivilegeManagerInterface $privilegeManager,
+        private readonly UserService $userService,
+        private readonly Connection $dbal,
+    ) {
+    }
+
+    public function getWorkspaceMetadata(ContentRepositoryId $contentRepositoryId, WorkspaceName $workspaceName): WorkspaceMetadata
+    {
+        $this->requireWorkspace($contentRepositoryId, $workspaceName);
+
+        $table = self::TABLE_NAME_WORKSPACE_METADATA;
+        $query = <<<SQL
+            SELECT
+                *
+            FROM
+                {$table}
+            WHERE
+                content_repository_id = :contentRepositoryId
+                AND workspace_name = :workspaceName
+        SQL;
+        $metadataRow = $this->dbal->fetchAssociative($query, [
+            'contentRepositoryId' => $contentRepositoryId->value,
+            'workspaceName' => $workspaceName->value,
+        ]);
+        if ($metadataRow === false) {
+            return new WorkspaceMetadata(
+                $workspaceName,
+                WorkspaceTitle::fromString(ucfirst($workspaceName->value)),
+                WorkspaceDescription::fromString(''),
+                $workspaceName->isLive() ? WorkspaceClassification::ROOT : WorkspaceClassification::UNKNOWN,
+            );
+        }
+        return new WorkspaceMetadata(
+            $workspaceName,
+            WorkspaceTitle::fromString($metadataRow['title']),
+            WorkspaceDescription::fromString($metadataRow['description']),
+            WorkspaceClassification::from($metadataRow['classification'])
+        );
+    }
+
+    public function getPersonalWorkspaceForUser(ContentRepositoryId $contentRepositoryId, UserId $userId): Workspace
+    {
+        $workspaceName = $this->findPrimaryWorkspaceNameForUser($contentRepositoryId, $userId);
+        if ($workspaceName === null) {
+            throw new \RuntimeException(sprintf('No workspace is assigned to the user with id "%s")', $userId->value), 1718293801);
+        }
+        return $this->requireWorkspace($contentRepositoryId, $workspaceName);
+    }
+
+
+
+    public function createRootWorkspace(
+        ContentRepositoryId $contentRepositoryId,
+        WorkspaceTitle $title,
+        WorkspaceDescription $description,
+    ): WorkspaceName
+    {
```

Comment:
```suggestion
    ): WorkspaceName {
```

Reply:
Thanks, but this is just a draft. Please don't add comments re cosmetics just yet